scrapy/downloadermiddlewares/retry.py

```diff
@@ -37,6 +38,13 @@ def EXCEPTIONS_TO_RETRY(cls):
             load_object(x) if isinstance(x, str) else x
             for x in Settings().getlist("RETRY_EXCEPTIONS")
         )
+    raise AttributeError(
+        f"{self.__class__.__name__!r} object has no attribute {name!r}"
+    )
```

Comment:
Hmm… When is this raised?

Reply:
When you try to access an attribute that doesn't exist, like when `__getattr__()` isn't defined.

Reply:
I was thinking to define super call here, but it only makes sense if there is multiple inheritance and 2+ superclasses redefine `getattr` which is not common in middlewares.

But making super call requires function definition to be inside class (to fill `super()` with correct arguments) or to pass correct class as first argument. Since the function is shared between different classes, I decided to just raise `AtrributeError` as super call will do in regular single inheritance situation (actually super call is really weird because neither `object` nor `type` have defined `getattr` and yet super call of `getattr` somehow works).